frontend/src/scenes/cohorts/CohortFilters/CohortCriteriaGroups.tsx

```diff
@@ -12,14 +12,15 @@ import { Lettermark, LettermarkColor } from 'lib/lemon-ui/Lettermark'
 import { alphabet } from 'lib/utils'
 import { cohortEditLogic, CohortLogicProps } from 'scenes/cohorts/cohortEditLogic'
 import { CohortCriteriaRowBuilder } from 'scenes/cohorts/CohortFilters/CohortCriteriaRowBuilder'
-import { criteriaToBehavioralFilterType, isCohortCriteriaGroup } from 'scenes/cohorts/cohortUtils'
+import { criteriaToBehavioralFilterType, isCohortCriteriaGroup, useIsReadonlyCohort } from 'scenes/cohorts/cohortUtils'
 
 import { AndOrFilterSelect } from '~/queries/nodes/InsightViz/PropertyGroupFilters/AndOrFilterSelect'
 
 export function CohortCriteriaGroups(logicProps: CohortLogicProps): JSX.Element {
     const logic = cohortEditLogic(logicProps)
     const { cohort } = useValues(logic)
```

Comment:
same everywhere else using this hook

Reply:
The reason I didn't put `readOnly` in `CohortEditLogic` because its value depends upon `isNewCohort` and `isEditingCohort`

![image](https://github.com/PostHog/posthog/assets/79001785/5b3704fd-692e-47bb-88d8-353bd89d23b8)

Reply:
... both of which are values derived from the `cohortEditLogic` ?

Reply:
this is precisely what selectors are for, check here: https://keajs.org/docs/core/selectors/ :)